ecl/eclagent/eclagent.cpp

```diff
@@ -1925,7 +1925,7 @@ void EclAgent::doProcess()
         }
         {
             MTIME_SECTION(queryActiveTimer(), "Process");
-            Owned<IEclProcess> process = loadProcess();
+            Owned<IEclProcess> process = new EclProcessEx(loadProcess());
```

Comment:
related to our previous offline discussion, I'm still not sure why IEclProcess needs to be wrapped...

where workflow is needed, eclagent calls:
EclAgent::runProcess-> WorkflowMachine::perform, which ultimately calls IEclProcess->perform in a couple of places (WorkflowMachine::performItem and WorkflowMachine::performItemParallel)

Is there any reason why they can't use a GlobalCodeContextExtra wrapper at that point, avoiding the need to wrap IEclProcess as well e.g.:

```
void WorkflowMachine::performItem(unsigned wfid, unsigned scheduledWfid)
{
#ifdef TRACE_WORKFLOW
    if(currentWfid)
        LOG(MCworkflow, "Branching from workflow item %u", currentWfid);
    LOG(MCworkflow, "Performing workflow item %u", wfid);
#endif
    wfidStack.append(currentWfid);
    wfidStack.append(scheduledWfid);
    currentWfid = wfid;
    currentScheduledWfid = scheduledWfid;
    timestamp_type startTime = getTimeStampNowValue();

    **GlobalCodeContextExtra gctx(ctx, wfid);**
    CCycleTimer timer;
    **process->perform(&gctx, wfid);**
    noteTiming(wfid, startTime, timer.elapsedNs());
    scheduledWfid = wfidStack.popGet();
    currentWfid = wfidStack.popGet();
    if(currentWfid)
    {
#ifdef TRACE_WORKFLOW
        LOG(MCworkflow, "Returning to workflow item %u", currentWfid);
#endif
    }
}
```

Reply:
I think that would work equally well, and probably simpler.

Reply:
There is a potential danger if any function does not use this context that the calling code will not get the workflow id.  E.g. The IcodeContext used within a graph.

Combined with your other comment about IGlobalCodeContext::getWorkflowId(), I wonder whether it should only appear in the IGlobalCodeContext.  That would still be accessible from the dfu functions if they indicated they required it.